unifi_discovery/__init__.py

```
@@ -146,6 +166,94 @@ class UnifiDevice:
+    # Group IPs by hw_addr
+    mac_to_ips: dict[str, list[str]] = {}
+    for ip, device in response_list.items():
+        if device.hw_addr is not None:
+            mac_to_ips.setdefault(device.hw_addr, []).append(ip)
+
+    # Nothing to deduplicate
+    if all(len(ips) <= 1 for ips in mac_to_ips.values()):
+        return response_list
+
+    to_remove: set[str] = set()
+    for mac, ips in mac_to_ips.items():
```

Ruff (0.15.9)

[warning] Loop control variable `mac` not used within loop body

Rename unused `mac` to `_mac`

(B007)